.github/workflows/docker.yml

```diff
@@ -0,0 +1,50 @@
+name: publish-docker
+
+on:
+  push:
+    paths-ignore:
+      - ".github/**"
+      - "docs/**"
+      - "resources/**"
+      - "benchmark/**"
+      - "tests/**"
+      - "**/*.md"
+    branches:
+      - main
+    tags:
+      - "v*.*.*"
+
+jobs:
+  publish_docker_image:
+    runs-on: [self-hosted, linux-3090]
+    environment: 'prod'
+    env:
+      TAG_PREFIX: "openmmlab/lmdeploy"
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Get lmdeploy commit id
+        run: |
+          export commit_id=$(git log --format="%H" -n 1)
+          echo "commit id = $commit_id"
+          echo "COMMIT_ID=$commit_id" >> $GITHUB_ENV
+      - name: Get lmdeploy version
+        if: startsWith(github.ref, 'refs/tags/') == false
```

Comment:
How to invoke this action?

Reply:
These two steps are used to determine the tag of built docker image, which would be either `openmmlab/lmdeploy:latest` or `openmmlab/lmdeploy:v0.x.x`

1.  When a PR is merged into main branch and  this step`startsWith(github.ref, 'refs/tags/') == false` is met
2. When a tag is pushed and this step`startsWith(github.ref, 'refs/tags/') == true` is met